lib/awful/layout/init.lua

```diff
@@ -109,6 +110,56 @@ function layout.set(_layout, t)
     tag.setproperty(t, "layout", _layout)
 end
 
+--- Get the layout parameters used for the screen
+--
+-- This should give the same result as "arrange", but without the "geometries"
+-- paramater, as this is computed during arranging
+--
+-- If "t" is given, "screen" is ignored, if none are given, the mouse screen is
+-- used
+--
+-- @param[opt] t The tag to query
+-- @param[opt] screen The screen
+-- @return A table with the workarea (x, y, width, height), the screen
+--   geometry (x, y, width, height), the clients, the screen and call
+--   the "geometries" table with client as keys and geometry as value
+function layout.parameters(t, screen)
+    if not screen then
+        screen = capi.mouse.screen
+    end
```

Comment:
I like my idea more (made as a comment on the commit; why does GitHub have so many different possibilities?). Just removing this part should be enough, because `tag.selected(nil)` does the right thing and uses `awful.screen.selected()` to get a screen.